File: data/derived_cot/rq3_prompting/results/translation/qwen/cpp_to_py/code_output/GomokuGame.py
class GomokuGame:
    def __init__(self, board_size):
        self.board_size = board_size
        self.board = [[' ' for _ in range(board_size)] for _ in range(board_size)]
        self.current_player = 'X'

    def make_move(self, row, col):
        if 0 <= row < self.board_size and 0 <= col < self.board_size and self.board[row][col] == ' ':
            self.board[row][col] = self.current_player
            self.current_player = 'O' if self.current_player == 'X' else 'X'
            return True
        return False

    def check_winner(self):
        directions = [(0, 1), (1, 0), (1, 1), (1, -1)]
        for i in range(self.board_size):
            for j in range(self.board_size):
                if self.board[i][j] != ' ':
                    for dx, dy in directions:
                        if self._check_five_in_a_row(i, j, dx, dy):
                            return self.board[i][j]
        return None

    def get_board(self):
        return self.board

    def _check_five_in_a_row(self, row, col, dx, dy):
        count = 1
        symbol = self.board[row][col]
        for step in range(1, 5):
            new_row = row + dx * step
            new_col = col + dy * step
            if (new_row < 0 or new_row >= self.board_size or 
                new_col < 0 or new_col >= self.board_size or 
                self.board[new_row][new_col] != symbol):
                return False
            count += 1
        return count == 5

File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/py_to_cpp/code_output/GomokuGame.cpp
#include <vector>

class GomokuGame {
public:
    GomokuGame(int board_size) : board_size(board_size), current_player('X') {
        board.resize(board_size, std::vector<char>(board_size, ' '));
    }

    bool make_move(int row, int col) {
        if (board[row][col] == ' ') {
            board[row][col] = current_player;
            current_player = (current_player == 'X') ? 'O' : 'X';
            return true;
        }
        return false;
    }

    char check_winner() {
        int directions[4][2] = {{0, 1}, {1, 0}, {1, 1}, {1, -1}};
        for (int row = 0; row < board_size; row++) {
            for (int col = 0; col < board_size; col++) {
                if (board[row][col] != ' ') {
                    for (int d = 0; d < 4; d++) {
                        if (_check_five_in_a_row(row, col, directions[d][0], directions[d][1])) {
                            return board[row][col];
                        }
                    }
                }
            }
        }
        return '\0';
    }

    bool _check_five_in_a_row(int row, int col, int dx, int dy) {
        int count = 1;
        char symbol = board[row][col];
        for (int i = 1; i < 5; i++) {
            int new_row = row + dx * i;
            int new_col = col + dy * i;
            if (new_row < 0 || new_row >= board_size || new_col < 0 || new_col >= board_size) {
                return false;
            }
            if (board[new_row][new_col] != symbol) {
                return false;
            }
            count++;
        }
        return count == 5;
    }

private:
    int board_size;
    std::vector<std::vector<char>> board;
    char current_player;
};